tracer/src/Datadog.Trace/Span.cs

```diff
@@ -462,5 +458,23 @@ internal void SetDuration(TimeSpan duration)
         {
             Duration = duration;
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
I assume this gives a small perf penalty when debug is enabled, but a small benefit when its not?

Reply:
yup